src/include/txn.i

```diff
@@ -1061,17 +1061,33 @@ __wt_txn_search_check(WT_SESSION_IMPL *session)
     return (0);
 }
 
+/*
+ * __txn_rollback_required --
+ *     Return write conflict.
+ */
+static inline int
+__txn_rollback_required(WT_SESSION_IMPL *session)
+{
+    WT_STAT_CONN_INCR(session, txn_update_conflict);
+    WT_STAT_DATA_INCR(session, txn_update_conflict);
+    return (__wt_txn_rollback_required(session, "conflict between concurrent operations"));
+}
+
 /*
  * __wt_txn_update_check --
  *     Check if the current transaction can update an item.
  */
 static inline int
-__wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
+__wt_txn_update_check(
+  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, bool check_onpage_value)
```

Comment:
We can just use cbt to indicate whether we need to check the onpage value. But I think it is not clear so I introduce an extra bool flag.